src/python/tests/fasta/test_fasta_prep_data.py

```diff
@@ -0,0 +1,138 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit testing of `ensembl.io.genomio.fasta.process` module.
+
+"""
+
+from contextlib import nullcontext as does_not_raise
+import filecmp
+from pathlib import Path
+from typing import ContextManager, List, Set
+
+import pytest
+
+import ensembl.io.genomio.fasta.process as FastaProcessing
+
+
+class TestFastaProcess:
+    """Tests for the Fasta module, submodule 'process'."""
+
+    tmp_dir: Path
+    data_dir: Path
+
+    @pytest.fixture(scope="class", autouse=True)
+    def setup(self, tmp_dir: Path, files_dir: Path):
+        """Loads necessary fixtures and values as class attributes."""
+        type(self).tmp_dir = tmp_dir
+        type(self).data_dir = files_dir / "process_fasta"
+
+    @pytest.mark.parametrize(
+        "input_fasta, input_gbff, expected_output_fasta, pep_mode",
+        [
+            (
+                "input.protein.fa.gz",
+                "input.gbff.gz",
+                "output.protein.fa",
+                True,
+            ),
+            (
+                "input.protein.fa.gz",
+                None,
+                "output.protein.fa",
+                True,
+            ),
+            (
+                "input.protein.fa.gz",
+                None,
+                "output.protein.fa",
+                False,
+            ),
+            ("input.fna.gz", None, "output.nuc.fna", False),
+            ("input.fna.gz", None, "output.nuc.fna", True),
+        ],
+    )
+    def test_fasta_prep(
+        self, tmp_path: Path, input_fasta: str, input_gbff: str, expected_output_fasta: str, pep_mode: bool
+    ) -> None:
+        """Tests the `process.prep_fasta_data()` function.
+
+        Args:
+            tmp_path: Where temporary files will be created.
+            input_fasta: Name of the fasta file with example input, in the test folder.
+            input_gbff: Name of the input GBFF example input, in the test folder.
+            expected_output_fasta: Name of the output fasta file with expected output, in the test folder.
+            pep_mode: Boolean flag to set processing in peptide mode.
+        """
```

Comment:
Whilst not mandatory, it is common to find any output and `expected` arguments at the end of the parameters:
```suggestion
    @pytest.mark.parametrize(
        "input_fasta, input_gbff, pep_mode, expected_output_fasta",
        [
            (
                "input.protein.fa.gz",
                "input.gbff.gz",
                True,
                "output.protein.fa",
            ),
            (
                "input.protein.fa.gz",
                None,
                True,
                "output.protein.fa",
            ),
            (
                "input.protein.fa.gz",
                None,
                False,
                "output.protein.fa",
            ),
            ("input.fna.gz", None, False, "output.nuc.fna"),
            ("input.fna.gz", None, True, "output.nuc.fna"),
        ],
    )
    def test_fasta_prep(
        self, tmp_path: Path, input_fasta: str, input_gbff: str, pep_mode: bool, expected_output_fasta: str
    ) -> None:
        """Tests the `process.prep_fasta_data()` function.

        Args:
            tmp_path: Where temporary files will be created.
            input_fasta: Name of the fasta file with example input, in the test folder.
            input_gbff: Name of the input GBFF example input, in the test folder.
            pep_mode: Boolean flag to set processing in peptide mode.
            expected_output_fasta: Name of the output fasta file with expected output, in the test folder.
        """
```

Reply:
> Whilst not mandatory, it is common to find any output and `expected` arguments at the end of the parameters:

Ahh ha, ok I wasn't aware. Thanks for point that out @JAlvarezJarreta 👍